core/templates/pages/contributor-dashboard-page/services/contribution-opportunities.service.ts

```diff
@@ -17,115 +17,153 @@
-
-angular.module('oppia').factory('ContributionOpportunitiesService', [
-  '$uibModal', 'ContributionOpportunitiesBackendApiService',
-  'UrlInterpolationService',
-  function(
-      $uibModal, ContributionOpportunitiesBackendApiService,
-      UrlInterpolationService) {
-    var reloadOpportunitiesEventEmitter = new EventEmitter<void>();
-    var removeOpportunitiesEventEmitter = new EventEmitter<void>();
-
-    var skillOpportunitiesCursor = null;
-    var translationOpportunitiesCursor = null;
-    var voiceoverOpportunitiesCursor = null;
-    var moreSkillOpportunitiesAvailable = true;
-    var moreTranslationOpportunitiesAvailable = true;
-    var moreVoiceoverOpportunitiesAvailable = true;
-
-    var _getSkillOpportunities = function(cursor) {
-      return ContributionOpportunitiesBackendApiService
-        .fetchSkillOpportunitiesAsync(
-          cursor).then(({ opportunities, nextCursor, more }) => {
-          skillOpportunitiesCursor = nextCursor;
-          moreSkillOpportunitiesAvailable = more;
-          return {
-            opportunities: opportunities,
-            more: more
-          };
-        });
-    };
-    var _getTranslationOpportunities = function(languageCode, cursor) {
-      return ContributionOpportunitiesBackendApiService
-        .fetchTranslationOpportunitiesAsync(
-          languageCode, cursor).then(({ opportunities, nextCursor, more }) => {
-          translationOpportunitiesCursor = nextCursor;
-          moreTranslationOpportunitiesAvailable = more;
+@Injectable({
+  providedIn: 'root'
+})
+export class ContributionOpportunitiesService {
+    skillOpportunitiesCursor = null;
+    translationOpportunitiesCursor = null;
+    voiceoverOpportunitiesCursor = null;
+    moreSkillOpportunitiesAvailable : boolean = true;
+    moreTranslationOpportunitiesAvailable: boolean = true;
+    moreVoiceoverOpportunitiesAvailable: boolean = true;
+
+    constructor(
+        private contributionOpportunitiesBackendApiService:
+        ContributionOpportunitiesBackendApiService,
+        private urlInterpolationService: UrlInterpolationService,
+        private ngbModal: NgbModal
+    ) {}
+
+  private _reloadOpportunitiesEventEmitter = new EventEmitter<void>();
+  private _removeOpportunitiesEventEmitter = new EventEmitter<void>();
+
+  private _getSkillOpportunities(cursor: string):
+      Promise<{ more: boolean; opportunities: SkillOpportunity[] }> {
+    return this.contributionOpportunitiesBackendApiService
+      .fetchSkillOpportunitiesAsync(cursor)
+      .then(({opportunities, nextCursor, more}) => {
+        this.skillOpportunitiesCursor = nextCursor;
+        this.moreSkillOpportunitiesAvailable = more;
+        return {
+          opportunities: opportunities,
+          more: more
+        };
+      });
+  }
+
+  private _getTranslationOpportunities(languageCode: string, cursor: string):
+   Promise<{ more: boolean; opportunities: ExplorationOpportunitySummary[] }> {
```

Comment:
Can you check if there is an interface for this? And also do it for all the other functions that you defined the type manually.